packages/fxa-auth-server/lib/geocoding.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Client,
+  PlaceType2,
+  Status,
+} from '@googlemaps/google-maps-services-js';
+import countries from 'i18n-iso-countries';
+import { ConfigType } from '../config';
+import { Logger } from 'mozlog';
+const config: ConfigType = require('../config').getProperties();
+
+export type ZipLocation = {
+  city: string;
+  country: string;
+  countryCode: string;
+  state: string;
+  stateCode: string;
+};
```

Comment:
I got this output format from `geodb.js`, but I'm happy to remove the unnecessary files. Do you know if we need the state code or name?

Reply:
Hey Reino; actually, since we're not entirely sure what consumers of this helper will need yet (we'll probably need to confirm with Finance if they need `state` and/or `stateCode`), we could greatly simplify this by returning everything, i.e. by returning a [`GeocodeResult`](https://github.com/googlemaps/google-maps-services-js/blob/f0c7455a46cd457a9f34f50206df4b13d9a69b42/src/common.ts#L1348), and having the caller destructure to get what it needs. Would that work?

Reply:
In this case I would actually recommend against doing that, since it still takes some work to get the desired output from the [`GeocodeResult`](https://github.com/googlemaps/google-maps-services-js/blob/f0c7455a46cd457a9f34f50206df4b13d9a69b42/src/common.ts#L1348).

For example, depending on the country and/or region, the `City` value might be either the `locality` or the `administrative_area_level_2`, or there might be no value. So I think it'd be better to keep this logic inside the class.

Note, I'm also assuming that this logic will change in future, as we start working with more and more data/countries.

Reply:
Okay thanks for the extra context. By the way I confirmed with Finance in https://mozilla-hub.atlassian.net/browse/FXA-4223?focusedCommentId=488672, and we only need `country` and `state` -- perhaps we can omit the other ones?

Reply:
Cool, I've removed the following `city`, `country` and `state`. So we only have `countryCode` and `stateCode` left.